crates/cargo-test-support/src/registry.rs

```diff
@@ -650,9 +698,139 @@ impl HttpServer {
 
     /// Route the request
     fn route(&self, req: &Request) -> Response {
-        let authorized = |mutatation: bool| {
-            if mutatation || self.auth_required {
-                self.token == req.authorization
+        pub struct Mutation<'a> {
+            pub mutation: &'a str,
+            pub name: Option<&'a str>,
+            pub vers: Option<&'a str>,
+            pub cksum: Option<&'a str>,
+        }
+        let authorized = |mutation: Option<Mutation>| {
+            if mutation.is_some() || self.auth_required {
+                match &self.token {
+                    Token::Bearer(token) => Some(token) == req.authorization.as_ref(),
+                    Token::Keys(private_key, private_key_subject) => {
+                        macro_rules! t {
```

Comment:
Consider factoring this out into a separate method for validating the PASETO

Reply:
done